module-gui/gui/core/DrawCommand.cpp

```diff
@@ -228,19 +228,26 @@ namespace gui
         uint32_t offsetRowContext = 0;
         uint32_t imageOffset      = 0;
         uint8_t alphaColor        = vecMap->getAlphaColor();
+        checkImageSize(ctx, vecMap);
 
         for (uint32_t row = 0; row < std::min(vecMap->getHeight(), ctx->getH()); row++) {
-            checkImageSize(ctx, vecMap);
+
             uint16_t vecCount = *(vecMap->getData() + imageOffset);
-            imageOffset += sizeof(uint16_t);
+            imageOffset += sizeof(uint8_t);
+            vecCount <<= 8;
+            vecCount |= *(vecMap->getData() + imageOffset);
+            imageOffset += sizeof(uint8_t);
```

Comment:
as exactly the same part is repeated 2x, maybe consider a function or at least a lambda?
e.g.
```C++
auto doTheMagic  = [&](){
    vecCount <<= 8;
    vecCount |= *(vecMap->getData() + imageOffset);
    imageOffset += sizeof(uint8_t);
};

```
and then just:
```C++
doTheMagic();
```

Reply:
and I believe the vecCount operations could be merged into one line - but it's just a cosmetic issue :P

Reply:
I merely made it verbose for clarity.  If there are no objections then I would prefer to use
```
uint16_t vecCount = (*(vecMap->getData() + imageOffset++) << 8) |
                     *(vecMap->getData() + imageOffset++);
```

Reply:
Actually, it's undefined behavior to post-increment a variable twice in a statement, so I would use this:
```
uint16_t vecCount = (*(vecMap->getData() + imageOffset++) << 8);
         vecCount |= *(vecMap->getData() + imageOffset++);
```

Reply:
to be 100% clear and transparent in this operation i'd cast the dereferenced pointer to 16-bit variable before shifting ;)

Reply:
Casting seemed a bit silly.  The current commit does this:
```
uint16_t vecOffset = vecMap->getData()[imageOffset++];
         vecOffset = (vecOffset << 8) | vecMap->getData()[imageOffset++];
```